gfxr_ext/decode/dive_vulkan_replay_consumer.cpp

```diff
@@ -699,5 +701,73 @@ void DiveVulkanReplayConsumer::ProcessFrameEndMarker(uint64_t frame_number)
     }
 }
 
+void DiveVulkanReplayConsumer::Process_vkGetFenceFdKHR(
+const ApiCallInfo&                                 call_info,
+VkResult                                           returnValue,
+format::HandleId                                   device,
+StructPointerDecoder<Decoded_VkFenceGetFdInfoKHR>* pGetFdInfo,
+PointerDecoder<int>*                               pFd)
+{
+    // Workaround for compositor captures. A fence may not have been created with the exportable
+    // flag, causing vkGetFenceFdKHR to fail on replay. Since the file descriptor is not used, we
+    // can ignore the error and allow replay to continue. The workaround tries with the call and if
+    // it failed, then ignore the error and continue the replay.
+    auto in_device = GetObjectInfoTable().GetVkDeviceInfo(device);
+    MapStructHandles(pGetFdInfo->GetMetaStructPointer(), GetObjectInfoTable());
+    auto in_pGetFdInfo = pGetFdInfo->GetPointer();
+    int* out_pFd = pFd->IsNull() ? nullptr : pFd->AllocateOutputData(1, -1);
+
+    VkResult replay_result = GetDeviceTable(in_device->handle)
+                             ->GetFenceFdKHR(in_device->handle, in_pGetFdInfo, out_pFd);
+
+    if (replay_result != VK_SUCCESS)
+    {
+        GFXRECON_LOG_WARNING("vkGetFenceFdKHR failed with error %s. Ignoring error and continuing "
+                             "replay.",
+                             util::ToString(replay_result).c_str());
+        replay_result = VK_SUCCESS;
+        return;
+    }
+
+    VulkanReplayConsumer::Process_vkGetFenceFdKHR(call_info, returnValue, device, pGetFdInfo, pFd);
```

Comment:
if the result is vk_success, we call GetFenceFdKHR again within Process_vkGetFenceFdKHR? would this cause problem?